dev/integration_tests/windows_startup_test/windows/runner/flutter_window.cpp

```diff
@@ -50,11 +65,29 @@ bool FlutterWindow::OnCreate() {
       &flutter::StandardMethodCodec::GetInstance());
 
   channel.SetMethodCallHandler(
-    [](const flutter::MethodCall<>& call,
+    [&](const flutter::MethodCall<>& call,
        std::unique_ptr<flutter::MethodResult<>> result) {
       std::scoped_lock lock(visible_mutex);
-      if (call.method_name() == "isWindowVisible") {
+      std::string method = call.method_name();
+
+      if (method == "isWindowVisible") {
         result->Success(visible);
+      } else if (method == "getWindowAttributeBool") {
```

Comment:
Given how specific this test is to dark mode, I'd just call this method `supportsDarkMode` and not bother passing the window attribute enum value across as a parameter but just hardcode it here. We're unlikely to ever expose the enum value to actual Dart code, no not something we probably need to explicitly test from the Dart side of the test.

Reply:
Done! I moved all Windows logic from the Dart layer down to the C++ layer.

(FWIW, I wrote it this way as eventually I'd like this to use FFI instead of method channels. The code was meant to be close to what the FFI version would be. Until then, it makes sense to keep things simple!)